src/hepattn/flex/local_ca.py

```diff
@@ -5,46 +5,44 @@
 create_block_mask: _mask_mod_signature = torch.compile(create_block_mask, dynamic=True)
 
 
+def _round_mul(q_idx, num, den):
+    """Compute round(q_idx * num / den) using integer arithmetic to avoid floating point operations."""
+    return (q_idx * num + (den // 2)) // den
+
+
 def sliding_window_mask_strided(
     window_size: int,
-    stride: float,
     q_len: int,
     kv_len: int,
     device: str,
 ) -> _mask_mod_signature:
     if window_size % 2 != 0:
         raise ValueError("Window size must be even for strided sliding window")
 
-    stride_val = torch.tensor(stride, device=device)
-
     def mask_mod(b, h, q_idx, kv_idx):  # noqa: ARG001
         # b = 0, h = 0 here
-        q_center = torch.round(q_idx * stride_val)
+        q_center = _round_mul(q_idx, kv_len, q_len)
         return (kv_idx - q_center).abs() <= window_size // 2
 
     return create_block_mask(mask_mod, B=None, H=None, Q_LEN=q_len, KV_LEN=kv_len, device=device)
 
 
 def sliding_window_mask_strided_wrapped(
     window_size: int,
-    stride: float,
     q_len: int,
     kv_len: int,
     device: str,
 ) -> _mask_mod_signature:
     if window_size % 2 != 0:
         raise ValueError("Window size must be even for strided sliding window")
 
-    stride_val = torch.tensor(stride, device=device)
-    kv_len_t = torch.as_tensor(kv_len, device=device).reshape(())
-
     def mask_mod(b, h, q_idx, kv_idx):  # noqa: ARG001
         # b = 0, h = 0 here
-        q_center = torch.round(q_idx * stride_val)
+        q_center = _round_mul(q_idx, kv_len, q_len)
 
         diagonal = (kv_idx - q_center).abs() <= window_size // 2
-        wrap_left = (kv_idx - q_center + kv_len_t).abs() <= window_size // 2
-        wrap_right = (kv_idx - q_center - kv_len_t).abs() <= window_size // 2
+        wrap_left = (kv_idx - q_center + kv_len).abs() <= window_size // 2
+        wrap_right = (kv_idx - q_center - kv_len).abs() <= window_size // 2
 
         return diagonal | wrap_left | wrap_right
```

Comment:
Can you check if this is faster (and produced the same results)?

```python
def mask_mod(b, h, q_idx, kv_idx):  # noqa: ARG001
    q_center = _round_mul(q_idx, kv_len, q_len)
    diff = (kv_idx - q_center).abs()
    return diff.min(kv_len - diff) <= window_size // 2
```
Even small gains in speed here should help